pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1620,8 +1621,72 @@ func (volTopology *wcpControllerVolumeTopology) GetTopologyInfoFromNodes(ctx con
 
 	switch strings.ToLower(params.StorageTopologyType) {
 	case "zonal":
-		// If the topology requirement received has just one zone, use the same zone as node affinity terms on PV.
-		if len(params.TopologyRequirement.GetPreferred()) == 1 {
+		if params.TopologyRequirement == nil {
+			// This case is for static volume provisioning using CNSRegisterVolume API
+			if !isPodVMOnStretchedSupervisorEnabled {
+				return nil, logger.LogNewErrorf(log, "topology requirement should not be nil. invalid params: %v", params)
+			} else {
+				// If the topology requirement received is nil, then identify topology of the datastore by looking into
+				// azClustersMap
+				var selectedSegments []map[string]string
+				for az, clusters := range azClustersMap {
+					var sharedDatastoresForclusterMorefs []*cnsvsphere.DatastoreInfo
+					for index, clusterMoref := range clusters {
```

Comment:
this "for" block is already present in GetSharedDatastoresInTopology() function, can create separate function to avoid code repetition

Reply:
done removed duplicate code in this commit https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/2581/commits/32a1d679d32787648d42e9470b7ba765241c6248